src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -467,94 +466,109 @@ bool PolicyManagerImpl::LoadPT(const std::string& file,
-      ForcePTExchange();
-      return false;
-    }
+  // Replace current data with updated
+  if (!cache_->ApplyUpdate(*pt_update)) {
+    LOG4CXX_WARN(
+        logger_,
+        "Unsuccessful save of updated policy table, trying another exchange");
+    return LoadPtResult::kNewPtRequired;
+  }
 
-    ExternalConsentStatus status = cache_->GetExternalConsentStatus();
-    GroupsByExternalConsentStatus groups_by_status =
-        cache_->GetGroupsWithSameEntities(status);
+  ExternalConsentStatus status = cache_->GetExternalConsentStatus();
+  GroupsByExternalConsentStatus groups_by_status =
+      cache_->GetGroupsWithSameEntities(status);
 
-    ProcessExternalConsentStatusUpdate(
-        groups_by_status, ConsentProcessingPolicy::kExternalConsentBased);
+  ProcessExternalConsentStatusUpdate(
+      groups_by_status, ConsentProcessingPolicy::kExternalConsentBased);
 
-    ProcessAppPolicyCheckResults(
-        results, pt_update->policy_table.app_policies_section.apps);
+  ProcessAppPolicyCheckResults(
+      results, pt_update->policy_table.app_policies_section.apps);
 
-    CheckPermissionsChangesAfterUpdate(*pt_update, *policy_table_snapshot);
+  CheckPermissionsChangesAfterUpdate(*pt_update, *policy_table_snapshot);
 
-    listener_->OnCertificateUpdated(
-        *(pt_update->policy_table.module_config.certificate));
+  listener_->OnCertificateUpdated(
+      *(pt_update->policy_table.module_config.certificate));
 
-    std::map<std::string, StringArray> app_hmi_types;
-    cache_->GetHMIAppTypeAfterUpdate(app_hmi_types);
-    if (!app_hmi_types.empty()) {
-      LOG4CXX_INFO(logger_, "app_hmi_types is full calling OnUpdateHMIAppType");
-      listener_->OnUpdateHMIAppType(app_hmi_types);
-    } else {
-      LOG4CXX_INFO(logger_, "app_hmi_types empty");
-    }
+  std::map<std::string, StringArray> app_hmi_types;
+  cache_->GetHMIAppTypeAfterUpdate(app_hmi_types);
+  if (!app_hmi_types.empty()) {
+    LOG4CXX_INFO(logger_, "app_hmi_types is full calling OnUpdateHMIAppType");
+    listener_->OnUpdateHMIAppType(app_hmi_types);
+  } else {
+    LOG4CXX_INFO(logger_, "app_hmi_types empty");
+  }
 
-    std::vector<std::string> enabled_apps;
-    cache_->GetEnabledCloudApps(enabled_apps);
-    for (auto it = enabled_apps.begin(); it != enabled_apps.end(); ++it) {
-      SendAuthTokenUpdated(*it);
-    }
+  std::vector<std::string> enabled_apps;
+  cache_->GetEnabledCloudApps(enabled_apps);
+  for (auto it = enabled_apps.begin(); it != enabled_apps.end(); ++it) {
+    SendAuthTokenUpdated(*it);
+  }
+
+  return LoadPtResult::kSuccess;
+}
+
+void PolicyManagerImpl::OnPTUFinished(const LoadPtResult ptu_result) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (LoadPtResult::kWrongPtReceived == ptu_result) {
+    LOG4CXX_DEBUG(logger_, "Wrong PT was received");
+    update_status_manager_.OnWrongUpdateReceived();
+    return;
+  }
+
+  update_status_manager_.OnValidUpdateReceived();
```

Comment:
Do we really need to call this if result is `LoadPtResult::kNewPtRequired` ? Since we'll schedule next PT update right away in that case

Reply:
@mked-luxoft yes, as SDL should send UP_TO_DATE before starting new PT interation. The same logic was in `LoadPT` before my fix